api/proto/metrics/metrics.proto

```diff
@@ -20,8 +20,40 @@ enum ZmetricTypes {
 message memoryMetric {
   uint32 usedMem = 2;            //in MBytes
   uint32 availMem = 3;           //in MBytes
-  double usedPercentage = 4;
-  double availPercentage = 5;
+  double usedPercentage = 4; // Obsolete - No Longer Used.
+  double availPercentage = 5; // Obsolete - No Longer Used.
+}
+
+// App Memory Metric
+message AppMemoryMetric {
+    // totalMB - Memory allocated to app instance by EVE. In Steady State,
+    //  this is expected to be same as in App instance config. Eve should
+    //  report what it actually allocated / reserved instead of what is configured.
+    uint32 totalMB = 1;
+
+    // usedMB - Memory Used by app from totalMB Memory
+    //  usedMB <= totalMB in normal cases.
+    //  But usedMB can exceed totalMB - this could be a warning to the user.
+    uint32 usedMB = 2;
+}
+
+message DeviceMemoryMetric {
+  // totalMB - Total Device memory as seen by EVE.
+  uint32 totalMB = 1;
+
+  // allocated_appsMB - Total memory Allocated to AppInstances
+  uint32 allocated_appsMB = 2;
+
+  // allocated_eveMB - Total Allocated Memory for Eve
+  // This is the Max available memory for EVE on the device.
+  // Eve may not be able to enforce this always. Controllers can flag
+  //    such cases as warnings to the user.
```

Comment:
Why describe controller policy here? 
It sufficies to say that the enforcement might be lax meaning that the used_eveMB can exceed the allocated_eveMB. BTW: no need to have that comment twice - sufficient to have it for the used_eveMB field.